### ISSUE TITLE
What's the meaning of the variable "rank" ?How could the deal with RGB images?

### ISSUE THREAD

**Issue by dazhenseu** [henrygouk/nnet#3]
Hi~ 
     Long time pasts before I getting this work started again:).
    I read your codes,it's writen in a simple and understandable way, thanks for sharing! 
    But I am not quite sure for the meaning of "rank" in "Convolutional". Does it equal to 2 when the original image is gray image or it equal to 3 when referring to RGB? And it's strange in the SpatialConvolutional class, the rank's value is hard coded with 2, so how could this class deal with RGB datasets like cifar10? Futher more, how could users deal with RGB datasets with other classes?
    I ran with cifar10 and set the rank to be 2, and found the accuracy is only around 10% .Of course,I just use the network in your mnist example, but I'm not sure which the reason is.

**Comment by henrygouk** [comment]
Hi,

The rank is the dimensionality of the signal you wish to operate on. I.e., 1 for audio, 2 for images, 3 for video, etc..

To work with RGB images you should pass 3 as the 'input' parameter, rather than 1.

I'm about to push a new CIFAR-10 example. Unfortunately I haven't implemented a lot of newer techniques, such as batch normalisation and the Adam optimisation method, so the network is very sensitive to hyperparameters which have not been tuned. In order to get anything resembling good performance you will have to tune the hyperparameters (mainly initial weight ranges).

**Comment by dazhenseu** [comment]
Thanks, your new example works. In order to get more satisfied results, I think the model may also need to be tuned. After all, for the cifar-10 dataset, it's not easy to get high performance.

**Comment by henrygouk** [comment]
Hi,

dd33ce4 just improved this example. The model is now a bit bigger, but results should be better.

You may want to try adding some dropout layers after the two hidden fully connected layers. Also, doing some data augmentation (e.g. training on horizontal flips) could be good, too.
